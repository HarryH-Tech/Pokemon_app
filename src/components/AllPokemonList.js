import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { getAllPokemonAction } from "../actions/AllPokemonList";

export function AllPokemonList(props) {
  console.log(props);

  const { allPokemon } = props;

  useEffect(() => {
    props.getAllPokemon();
  }, []);

  console.log(allPokemon);

  return (
    <div>
      <h2>POKEMON LIST</h2>
      {allPokemon
        ? allPokemon.map((pokemon) => {
            return <p>{pokemon.name}</p>;
          })
        : ""}
    </div>
  );
}

const mapStateToProps = (state) => {
  console.log(state);
  return {
    allPokemon: state.allPokemon,
  };
};

const mapDispatchToProps = {
  getAllPokemon: getAllPokemonAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(AllPokemonList);
