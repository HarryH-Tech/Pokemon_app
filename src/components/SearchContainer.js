import React, { useState } from "react";
import { connect } from "react-redux";
import { searchPokemonAction } from "../actions/SearchPokemon";

export function SearchContainer(props) {
  // console.log(props);
  const [searchInput, setSearchInput] = useState("bulbasaur");

  const handleChange = (e) => {
    setSearchInput(e.target.value);
    console.log(searchInput);
  };

  const handleSearch = () => {
    console.log("SEARCHING", searchInput);
    props.searchPokemon(searchInput);
  };

  return (
    <div>
      <input name="searchInput" onChange={handleChange} />
      <button onClick={handleSearch}>Search</button>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    pokemon: state,
  };
};

const mapDispatchToProps = {
  searchPokemon: searchPokemonAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchContainer);
