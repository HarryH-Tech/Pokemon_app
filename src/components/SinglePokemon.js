import React, { useEffect } from "react";
import { connect } from "react-redux";
import { getSinglePokemonAction } from "../actions";

function SinglePokemon(props) {
  console.log(props.match.params.name);
  console.log(props);

  useEffect(() => {
    props.getSinglePokemon(props.match.params.name);
  }, []);

  return <div>POKEMON</div>;
}

const mapStateToProps = (state) => {
  return {
    singlePokemon: state,
  };
};

const mapDispatchToProps = {
  getSinglePokemon: getSinglePokemonAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(SinglePokemon);
