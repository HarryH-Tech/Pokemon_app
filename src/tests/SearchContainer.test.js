import React from "react";
import { shallow } from "enzyme";
import { SearchContainer } from "../components/SearchContainer";

describe("Search Container", () => {
  let searchContainer = shallow(<SearchContainer />);

  it("renders successfully", () => {
    expect(searchContainer).toMatchSnapshot();
  });

  //   it("Contains the List component", () => {
  //     expect(app.find("Connect(AllPokemonList)").exists()).toBe(true);
  //   });
});
