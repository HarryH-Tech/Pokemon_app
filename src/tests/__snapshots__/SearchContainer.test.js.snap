// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Container renders successfully 1`] = `ShallowWrapper {}`;
